pkg/agent/agent_test.go

```diff
@@ -140,9 +183,25 @@ func TestListReleases(t *testing.T) {
 				t.Errorf("%v", err)
 			}
 
+			// Check for size of returned apps
 			if got, want := len(apps), len(tc.expectedApps); got != want {
 				t.Errorf("got: %d, want: %d", got, want)
 			}
+
+			//Check if expected and returned contain the same releases, given they have same size
+			m := make(map[releaseStub]bool)
+
+			for _, eapp := range tc.expectedApps {
+				m[releaseStub{eapp.ReleaseName, eapp.Namespace, strtoi(eapp.Version)}] = true
+			}
+
+			for _, app := range apps {
+				rs := releaseStub{app.ReleaseName, app.Namespace, strtoi(app.Version)}
```

Comment:
So I think this is comparing a manually constructed `releaseStub` made from the output `AppOverview`, with another manually constructed `releaseStub` created from the expected `AppOverview`. Why not just compare the output `AppOverview` with the expected `AppOverview` directly? (bonus that they then have the same types).

Also, not sure if it could have been used here, but the `cmp` from the `go-cmp` module is quite nice for comparisons... you might also be able to do something like:
```
import "github.com/google/go-cmp/cmp"
...
    if !cmp.Equal(tc.expectedApps, apps) {
        t.Errorf(cmp.Diff(tc.expectedApps, apps))
    }
```

Reply:
Absolutely, comparing `AppOverView`'s makes more sense.
I'll take a look at `go-cmp`, but I have a gut feeling that additional fields need to be filled in the expected `AppOverview` array.

I'll create a new PR since this one is merged & closed.

Reply:
I tried that while reviewing this PR and it doesn't work (that's why I assumed it was doing the comparison manually). There are still some fields that are not in the expected (like the `status` of the release) and also the order in the slices changes sometimes so the comparison fails.

Reply:
I was able to fix it by using the library @absoludity mentioned in PR #1379.

It was done by mapping a unique id (presumably no two releases can have the same name in the same namespaces in the expected results) to a pointer to the expected `AppOverview`.

Then, by generating a unique id for the attained result, looking up it in the mapping, and then and only then, perform deep equality check on both objects.

Reply:
Looks like it got even simpler than that too. Great, thanks @latiif